### PR TITLE
Fetch cached gene-panel-data from s3 if available

### PR DESCRIPTION
**Pull request by alisman**
Try fetching gene_panel_data from s3 and go to web server if it doesn't exist

### CODE REVIEW

File: packages/cbioportal-frontend-commons/src/lib/apiClientCache.ts [hunk 1]
```diff
@@ -128,6 +129,39 @@ export function cachePostMethod(
...
+
+    if (methodName in cachedMethods) {
+        const oldMethod = targetObj[methodName];
+
+        targetObj[methodName] = function(arg: any) {
+            const hash = getHash(arg);
+            const args = arguments; // keep them locally for closure
+            const fileName = `${methodName}-${hash}.json`;
+
+            return $.get(
```
Comment (alisman) [inline packages/cbioportal-frontend-commons/src/lib/apiClientCache.ts:151]: i got sick of wrestling with fetch api which drives me nuts.

File: packages/cbioportal-frontend-commons/src/lib/apiClientCache.ts [hunk 2]
```diff
@@ -142,13 +176,23 @@ export function cachePostMethodsOnClient(
...
             obj.prototype,
             methodName,
             apiCacheLimit,
             log,
             sentryLog
-        )
-    );
+        );
+
+        if (window.location.hostname.includes('genie-private')) {
```
Comment (alisman) [inline packages/cbioportal-frontend-commons/src/lib/apiClientCache.ts:188]: we can expand this as necessary.  obviously is this temporary solution

**Review by adamabeshouse — APPROVED** [review]
interesting

File: packages/cbioportal-frontend-commons/src/lib/apiClientCache.ts (hunk 2 above)
Comment (inodb) [reply packages/cbioportal-frontend-commons/src/lib/apiClientCache.ts:199]: can we include `genie.cbioportal.org` as well? For `genie_public`

**Review by inodb — CHANGES_REQUESTED** [review]
seeing some strange errors after adding some filters: https://bit.ly/3gQyJyW
 
Screen Shot 2021-04-27 at 4 18 47 PM

EDIT: looks like this is by design (a 403 will result in calling the actual endpoint instead of getting the data from S3)

**Pull request merged**